### PR TITLE
Replace PHPSpec with PHPUnit

### PR DESCRIPTION
**Pull request by ruudk**
PHPSpec blocks the upgrade to Symfony 6. Replacing it with the industry standard PHPUnit.
This makes it way easier for contributors to work on the code.

Also dropped Scrutinizer.

See #17

### CODE REVIEW

File: tests/Unit/StopwatchPluginTest.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    public function setUp(): void
+    {
+        $this->stopwatch = new Stopwatch();
+        $this->plugin = new StopwatchPlugin($this->stopwatch);
+    }
+
+    /**
+     * @test
+     */
+    public function it_records_event(): void
```
Comment (dbu) [inline tests/Unit/StopwatchPluginTest.php:39]: the phpunit convention is to have methods start with `test` and then we would not need the annotation.

i don't know how much people use this form or the other form (but all things i work on have the `testXY` form). wdyt of renaming this to testRecordEvent, and the other testRecordError ?

**Review by dbu — COMMENTED** [review]
thanks, looks good! one suggestion about the test method names.

i wonder why the build shows a scrutinizer build when you are deleting that in the PR :thinking: but we can merge and then see if it goes away.

File: tests/Unit/StopwatchPluginTest.php (hunk 1 above)
Comment (ruudk) [reply tests/Unit/StopwatchPluginTest.php:39]: both are fine.. in our code base we only use it_should_be.... together with @test 
I decided to keep the original names from phpspec... given there are  just 2, does it really matter? :)

**Comment by ruudk** [comment]
the scrutinizer build failure is because of required branch checks. Can be removed after merging the pr.

edit: seems like it's not required... would be nice to setup requirements btw.

File: tests/Unit/StopwatchPluginTest.php (hunk 1 above)
Comment (dbu) [reply tests/Unit/StopwatchPluginTest.php:39]: no, does not really matter. just noticed and wanted to discuss it. fine for me.

**Comment by dbu** [comment]
thanks @ruudk !

**Comment by Nyholm** [comment]
Thank you both!

**Pull request merged**